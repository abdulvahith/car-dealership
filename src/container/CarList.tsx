import React, { useEffect, useState } from "react";
import Box from "@mui/material/Box";
import { data } from "../constants/appConstants";
import { useParams } from "react-router-dom";
import Breadcrumbs from "@mui/material/Breadcrumbs";
import Snackbar from "@mui/material/Snackbar";
import IconButton from "@mui/material/IconButton";
import NavigateBeforeIcon from "@mui/icons-material/NavigateBefore";
import { Link } from "@mui/material";
import { Cars, CarsBrand } from "./types";
import { Toastr, CarListHeader } from "./style";
import CarListComp from "../components/CarListComp";

const CarList: React.FC<any> = () => {
  const { id } = useParams();
  const [carDetails, setCarDetails] = useState<Cars[]>([]);
  const [showSuccess, setShowSuccess] = useState<boolean>(false);

  useEffect(() => {
    const res: CarsBrand = data.find((cars) => cars.id === id) as CarsBrand;
    setCarDetails(res?.carList);
  }, [id]);

  return (
    <>
      <div
        role="presentation"
        onClick={() => undefined}
        style={{
          position: "relative",
        }}
      >
        <Breadcrumbs aria-label="breadcrumb">
          <Link underline="hover" href="/" color="black">
            <IconButton aria-label="Example">
              <NavigateBeforeIcon />
            </IconButton>
            <span
              style={{
                position: "absolute",
                top: "10px",
              }}
            >
              Select Brand
            </span>
          </Link>
        </Breadcrumbs>
      </div>
      {showSuccess && (
        <Snackbar
          open={showSuccess}
          autoHideDuration={6000}
          anchorOrigin={{ vertical: "top", horizontal: "right" }}
          onClose={() => setShowSuccess(false)}
        >
          <Toastr onClose={() => setShowSuccess(false)} severity="success">
            Our team will contact you soon!
          </Toastr>
        </Snackbar>
      )}

      <Box
        sx={{
          bgcolor: "background.paper",
          pt: 8,
          pb: 6,
        }}
      >
        <CarListHeader variant="h4" align="left" color="text.primary">
          Select {id} car below
        </CarListHeader>
      </Box>
      <CarListComp carDetails={carDetails} setShowSuccess={setShowSuccess} />
    </>
  );
};

export default CarList;
