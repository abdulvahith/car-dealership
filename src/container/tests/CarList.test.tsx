import React from "react";
import renderer from "react-test-renderer";
import CarList from "../CarList";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Enzyme from "enzyme";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

Enzyme.configure({ adapter: new Adapter() });

describe("CarListComp component should render properly", () => {
  test("renders without crashing", () => {
    render(
      <Router>
        <CarList />
      </Router>
    );
  });

  it("CarList renders correctly snapshot test", () => {
    const rendered = renderer.create(
      <Router>
        <CarList />
      </Router>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
