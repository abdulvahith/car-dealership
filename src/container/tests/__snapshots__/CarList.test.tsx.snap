// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarListComp component should render properly CarList renders correctly snapshot test 1`] = `
Array [
  <div
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <nav
      aria-label="breadcrumb"
      className="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        className="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          className="MuiBreadcrumbs-li"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-16cd5xe-MuiTypography-root-MuiLink-root"
            href="/"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <button
              aria-label="Example"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
            <span
              style={
                Object {
                  "position": "absolute",
                  "top": "10px",
                }
              }
            >
              Select Brand
            </span>
          </a>
        </li>
      </ol>
    </nav>
  </div>,
  <div
    className="MuiBox-root css-10m0cap"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft sc-gsDKAQ FGXDW css-fokda-MuiTypography-root"
    >
      Select 
       car below
    </h4>
  </div>,
]
`;
